nuitka/freezer/DllDependenciesPosix.py

```diff
@@ -44,32 +43,35 @@ def detectBinaryPathDLLsPosix(dll_filename, package_name, original_dir):
     # on Travis. pylint: disable=global-statement
     global _detected_python_rpath
     if _detected_python_rpath is None and not isPosixWindows():
-        _detected_python_rpath = getSharedLibraryRPATH(sys.executable) or False
+        _detected_python_rpath = getSharedLibraryRPATHS(sys.executable) or []
 
         if _detected_python_rpath:
             # Need to resolve a potential symlink.
             if os.path.islink(sys.executable):
-                sys_executable = resolveSymlink(sys.executable)
+                sys_executable = os.path.realpath(sys.executable)
             else:
                 sys_executable = sys.executable
 
-            _detected_python_rpath = _detected_python_rpath.replace(
-                "$ORIGIN", os.path.dirname(sys_executable)
-            )
+            for i, rpath in enumerate(_detected_python_rpath):
```

Comment:
I think I will strive to make this resolution a function, this happens in at least one other place too, and then we can just use a list contraction and don't need to modify the list.